aws/resource_aws_ssm_document.go

```diff
@@ -172,6 +172,12 @@ func resourceAwsSsmDocument() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 			},
+			"version_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
```

Comment:
Oddly enough, the [`UpdateDocument` API operation](https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_UpdateDocument.html) also accepts `VersionName`, so going to verify its behavior on merge.